dl_binaries.sh

```diff
@@ -2,21 +2,41 @@
 set -e
 
 # This script downloads the binaries for the most recent version of TabNine.
-
-version="$(curl -sS https://update.tabnine.com/bundles/version)"
-targets='x86_64-apple-darwin
+# Infrastructure detection heavily inspired by https://github.com/tzachar/cmp-tabnine/blob/main/install.sh
+version=${version:-$(curl -sS https://update.tabnine.com/bundles/version)}
+if [ $# -gt 0 ]; then
+    # Pass fully qualified names as positional arguments
+    targets="$(printf '%s\n' "$@")"
+else
+    case $(uname -s) in
+    "Darwin")
+        if [ "$(uname -m)" = "arm64" ]; then
+            targets="aarch64-apple-darwin"
+        else
+            targets="$(uname -m)-apple-darwin"
+        fi
+        ;;
+    "Linux")
+        targets="$(uname -m)-unknown-linux-musl"
```

Comment:
this won't work as if it's arm/linux isn't supported :(

Reply:
Indeed, good catch. It can now only catch the three supported versions, and falls back to a download of all three in other cases. (perhaps should give error message to user and halt rather than downloading all available binaries?)